src/app/ReadClient.cpp

```diff
@@ -466,7 +466,10 @@ CHIP_ERROR ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchange
 
     if ((!IsSubscriptionType() && !mPendingMoreChunks) || err != CHIP_NO_ERROR)
     {
-        Close(err);
+        // If we're just setting up this subscription and we received an error, don't try
+        // to reconnect, just error out.
+        bool reconnect = !(IsSubscriptionType() && IsAwaitingInitialReport());
+        Close(err, reconnect);
```

Comment:
No, the expectation is that _if_ resubscribe is enabled this code will try to resubscribe.  Consumers can cancel that themselves based on whatever logic they want to use.  This is totally going to break existing SDK consumers.

If the Pythoon test does not want a resubscribe, it should either cancel the resubscribe attempt or not be setting up auto-resubscribe to start with.